main.py

```diff
@@ -1,410 +1,69 @@
-				offset = obstacle.x - player.x, obstacle.y - player.y
-				boolean = player_mask.overlap(obstacle_mask, offset)
-				if boolean:
-					return True
-	return False
-
-
-def find_free_zone_y():
-	"""
-	To determine free space inorder to place coins. To prevent coins being drawn over obstacles.
-	"""
-	y_coordinate_obstacle1 = 310
-	y_coordinate_obstacle2 = 150
-	free_zone_y = max_y_coord
-
-	for obstacle in Tree.obstacles:
-		if obstacle.x >(bg.get_width() - obstacle.width) and obstacle.x < bg.get_width():
-			free_zone_y =  y_coordinate_obstacle1
-	for rock in Other_obstacles.obstacles:
-		if rock.x >(bg.get_width() - rock.width) and rock.x < bg.get_width():
-			free_zone_y =  y_coordinate_obstacle2
-
-	return free_zone_y
-
-def create_coin():
-	"""
-	Creates a coin in the free space. 
-	"""
-	free_zone_y = find_free_zone_y()	# find free space in y axis
-	x = random.randint(50,free_zone_y)	# choose random y value within free zone
-	coins.append(Coin(bg.get_width(), x))
-
-def update_coins_position():
-	"""
-	Updates the x coordinates of coins. If coin goes offscreen, remove it from the list.
-	"""
-	for coin in coins:
-		coin_width = coin.imgs[0].get_width()
-		if coin.x < -1*coin_width: # If coin goes offscreen, removing it from coins list 
-			coins.remove(coin)
-		else:
-			coin.x -= foreground_speed
-
-def coin_collection():
-	"""
-	Collision with coin is checked using Pixel perfect collision method. If collision occurs returns True, else False.
-	Collision is check only if coin is near the player to save computation.
-	"""
-	player_mask = pygame.mask.from_surface(player.img)
-	for coin in coins:
-		try:
-			if coin.x < (player.x + player.img.get_width()+10):	# Checking for collision if near player
-				coin_mask = pygame.mask.from_surface(coin.img)
-				offset = coin.x - player.x, coin.y - player.y
-				boolean = player_mask.overlap(coin_mask, offset)
-				if boolean:
-					coins.remove(coin)
-					return True
-		except: pass
-	return False
-
-def display_num_coins_collected():
-	"""
-	To display the number of coins collected.
-	"""
-	win.blit(coin_board, (10,10))
-	font = pygame.font.Font('freesansbold.ttf', 40)
-	text_x, text_y = 80, 20
-	text = font.render(str(num_coins_collected), True, (255,255,255))
-	win.blit(text, (text_x, text_y))
-
-
-def display_mouse_pointer_coordinates(mx,my):
-	"""
-	To display mouse pointer coordinates just for reference. Temporary function to be removed later.
-	"""
-	font = pygame.font.Font('freesansbold.ttf', 32)
-	text_x, text_y = 1400, 10
-	text = font.render(str(mx)+', '+str(my), True, (0,0,0))
-	win.blit(text, (text_x, text_y))
-
-def display_collision_message():
-	"""
-	Temporary function to communicate that collision had occured
-	"""
-	font = pygame.font.Font('freesansbold.ttf', 100)
-	text_x, text_y = 750, 100
-	text = font.render('COLLISION', True, (0,0,0))
-	win.blit(text, (text_x, text_y))
-
-# MAIN ALGORITHM
+# Global variables
+speed = 60		# fps

```

Comment:
In the future (mostly during Beta release), the user should be given an option to play the game in 30 fps or 60 fps. For now, the user can run 60 fps. Once the CV algorithms kick in, your game will lag!